mainapp/src/Admin.tsx

```diff
@@ -25,13 +25,37 @@ const style = {
 
 };
 
+type EditionTab
+    = 'UsersEdition' | 'ProfilesEdition' | 'SourcesEdition'
+
+const editionTabToNumber = (editionTab: EditionTab) => {
+    switch (editionTab) {
+        case 'UsersEdition': return 0
+        case 'ProfilesEdition': return 1
+        case 'SourcesEdition': return 2
+        default: 0
+    }
+}
+
+const editionTabToString = (editionTab: number): EditionTab => {
+    switch (editionTab) {
+        case 0: return 'UsersEdition'
+        case 1: return 'ProfilesEdition'
+        case 2: return 'SourcesEdition'
+        default: return 'UsersEdition'
+    }
+}
+
+
 type Model = {
     users: RD<string, User[]>,
     profiles: RD<string, string[]>,
-    isModalOpen: boolean
+    isModalOpen: boolean,
+    currentEditionTab: EditionTab
```

Comment:
I am not really comfortable with this. Store the opened tab id in the global model, this sounds a bit overkill. But this is maybe because I am not so familiar with this architecture.

Reply:
Since tabs are used as a proxy for routing, doesn't make sense to define the state at the top level ? child component such as Dispatcher use it to display the correct interface